src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1740,10 +1740,20 @@ private void CheckContainsDocument(DocumentId documentId)
             throw new ArgumentNullException(nameof(documentId));
         }
 
-        if (!ContainsDocument(documentId))
+        // For source generated documents we expect them to be already generated to use any of the APIs that call this
+        if (documentId.IsSourceGenerated &&
+            ContainsProject(documentId.ProjectId) &&
+            this.GetRequiredProject(documentId.ProjectId).TryGetSourceGeneratedDocumentForAlreadyGeneratedId(documentId) is not null)
```

Comment:
if it is SG and we got this call, don't we want to throw if the SG doc was not already generated?  

unclear why we're just basically ignoring this.

Reply:
We do throw. This is checking if the doc is _not_ null, and so returning, which means a valid check. If the doc wasn't generated it would fall through to the throw.

I got confused by this method at first too.